### PR TITLE
Fix errors in realPath arising from the string changes on Tuesday

### PR DESCRIPTION
**Pull request by lydia-duncan**
Something about the way the C function realpath creates the buffer it returns
was reacting poorly with gasnet fast and the new string set up.  Since the
locale method cwd() did not experience the same error, I surmised that it was
better to follow cwd()'s buffer creation strategy and use qio_malloc to create
a buffer to send to realpath.  This resolved the errors I was seeing.

Also added a free call to the error case for cwd, and made the string
construction in cwd identical to that in realpath

I'm not entirely certain why our string changes tickled this, others might have
a better idea.

### CODE REVIEW

**Comment by lydia-duncan** [comment]
@Kyle-B, care to review?

**Comment by lydia-duncan** [comment]
Verified that the gasnet fast failures now pass

**Comment by lydia-duncan** [comment]
Question for you: the call to create a new string in realPath passes different values as its second and third arguments, while the call to create a new string in locale.cwd() passes the same value as its second and third argument.  This was not the source of the error, but is it still something we should be concerned about?

**Comment by lydia-duncan** [comment]
std/ linux passed

File: runtime/src/chpl-file-utils.c [hunk 1]
```diff
@@ -332,11 +332,17 @@ qioerr chpl_fs_mkdir(const char* name, int mode, int parents) {
...
-    // If an error occurred, shortened will be NULL.  Otherwise, it will
-    // contain the cleaned up path.
+  size_t bufsize = MAXPATHLEN*sizeof(char);
+  char* bufptr;
+  char* pathbuf = (char *)qio_malloc(bufsize);
+
+  bufptr = realpath(path, pathbuf);
+  if (bufptr == NULL) {
+    // If an error occurred, bufptr will be NULL.  Otherwise, it will
+    // point to pathbuf anyways
```
Comment (Kyle-B) [inline runtime/src/chpl-file-utils.c]: You need to free `pathbuf` in this case.

**Comment by lydia-duncan** [comment]
Tested and pushed!  Thoughts on the module question?

**Comment by Kyle-B** [comment]
Re locale.cwd vs realPath:

Those are the length of the string and the size of the buffer.  Adding 1 is the correct way to do it. Not adding 1 will result in incorrect behavior in a few edge cases I believe (trying to call .c_str on it might not result in a null terminated string). I've been meaning to rename that constructor to something else, its not very safe and clearly pretty easy to mess up.

**Comment by lydia-duncan** [comment]
Okay, I'll fix that, too

**Comment by lydia-duncan** [comment]
Good to go?

**Comment by Kyle-B** [comment]
Seems fine to me.

**Pull request merged**